source/dofs/dof_handler.cc

```diff
@@ -3067,7 +3074,7 @@ DoFHandler<dim, spacedim>::connect_to_triangulation_signals()
 
       // refinement signals
       this->tria_listeners_for_transfer.push_back(
-        this->tria->signals.pre_distributed_refinement.connect(
+        this->tria->signals.post_p4est_refinement.connect(
```

Comment:
I don't understand why we need a new signal? It's not clear to me what's the difference between `pre_distributed_refinement` and `post_p4est_refinement`

Reply:
Long story short: p4est refines differently than Triangulation. We have functions that rely on correctly set refine flags, and the outcome of these functions affect partitioning. So we need this additional signal after refining p4est, but before repartitioning.

Maybe it gets clearer in context of the follow-up PR #11981.

Reply:
Yes, I understand that but you are not adding a signal here, you are simply replacing the signal by a new one and the documentation of these two signals is almost identical. Why can't you use the current `pre_distributed_refinement`?

Reply:
The connected function `pre_distributed_transfer_action()` stores all future FE indices in a container and prepares the transfer of them across mesh refinement. Right now, we could attach this signal to either `pre_distributed_refinement` or `post_p4est_refinement` and the outcome would be the same.

However, if we want to manipulate these future FE indices based on the way p4est manages mesh refinement (see #11981), we need to attach this function to the `post_p4est_refinement` signal.

I replaced the signal in this PR already to make sure that the testsuite succeeds with this change.

Reply:
Sorry for the confusion :)